### PR TITLE
[SYCL] Link sycl-devicelib-host lib in clang-cl + fscyl by default.

### PR DESCRIPTION
**Pull request by jinge90**
Signed-off-by: jinge90 <ge.jin@intel.com>

### CODE REVIEW

**Comment by jinge90** [comment]
Hi, @mdtoguchi 
Previously, we add sycl-devicelib-host.a in link list by default in if users are using "clang++ -fsycl". This default linked static library consists of implementation of imf libdevice functions for host device. However, we missed adding it in link list in clang-cl mode, this PR adds this static library if users are using "clang-cl -fsycl".

Thanks very much.

File: clang/lib/Driver/ToolChains/MSVC.cpp [hunk 1]
```diff
@@ -142,6 +142,9 @@ void visualstudio::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     CmdArgs.push_back("-defaultlib:sycl-devicelib-host.lib");
   }
 
+  if (C.getDriver().IsCLMode() && Args.hasArg(options::OPT_fsycl))
+    CmdArgs.push_back("-defaultlib::sycl-devicelib-host.lib");
+
```
Comment (mdtoguchi) [inline clang/lib/Driver/ToolChains/MSVC.cpp:147]: Changes are good, can we add a test to check as well?

Comment (jinge90) [reply clang/lib/Driver/ToolChains/MSVC.cpp:147]: Hi, @mdtoguchi 
Test is added, could you review again?
Thanks very much.

Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/MSVC.cpp:147]: Is there any particular reason why this lib is being added to the `link` command line and not being added to the default directives of the generated host object via `--dependent-lib=sycl-device-lib-host`?

Comment (jinge90) [reply clang/lib/Driver/ToolChains/MSVC.cpp:147]: Hi, @mdtoguchi 
Just checked the code, “dependent-lib=sycl-devicelib-host” is the right approach, I will update the patch.

Thanks very much.

**Review by mdtoguchi — APPROVED** [review]
Thanks!  LGTM

**Comment by jinge90** [comment]
> Thanks! LGTM

Hi, @mdtoguchi 
Sorry for bothering you again, I carelessly clicked the button and missed your approval.

**Review by mdtoguchi — APPROVED** [review]

**Comment by jinge90** [comment]
Hi, @tfzhu 
It seems that the failure from "SYCL / Linux / CUDA LLVM Test Suite (pull_request)" is not related to my patch, it this a known issue?
Thanks very much.

**Comment by tfzhu** [comment]
> Hi, @tfzhu It seems that the failure from "SYCL / Linux / CUDA LLVM Test Suite (pull_request)" is not related to my patch, it this a known issue? Thanks very much.

I have no idea. Maybe @bader or @pvchupin can give some comments.

**Comment by bader** [comment]
> Hi, @tfzhu It seems that the failure from "SYCL / Linux / CUDA LLVM Test Suite (pull_request)" is not related to my patch, it this a known issue? Thanks very much.

@jinge90, if you can't find it in https://github.com/intel/llvm/issues, please, open a new issue.

**Comment by jinge90** [comment]
Hi, @pvchupin 
All tests pass, could you help merge?

Thanks very much.

**Pull request merged**